### PR TITLE
Nomad Tests and libs

### PR DESCRIPTION
**Pull request by dbhatnagar-px**
Adds few more Nomad tests and configurations and supports Jenkins

### CODE REVIEW

File: tests/basic/nomad_test.go [hunk 1]
```diff
@@ -413,3 +466,798 @@ var _ = Describe("{RunFioJobWithSnapshotOperationsOnNomad}", func() {
...
+				err := client.DeleteJob(jobID)
+				if err != nil {
+					log.Errorf("Retry error: failed to delete Fio job %v: %v", jobID, err)
+					return nil, true, err
+				}
+				return nil, false, nil
+			}, 2*time.Minute, 20*time.Second)
+
+			if err != nil {
+				log.Errorf("Final error: failed to delete Fio job %v: %v", jobID, err)
```
Comment (lsrinivas-pure) [inline tests/basic/nomad_test.go:560]: don't you want to test to fail here?

File: tests/basic/nomad_test.go [hunk 2]
```diff
@@ -413,3 +466,798 @@ var _ = Describe("{RunFioJobWithSnapshotOperationsOnNomad}", func() {
...
+		err = client.CreateVolume(volumeID, pluginID, capacityMin, capacityMax, accessMode, attachmentMode, fsType, mountFlags, parameters)
+		log.FailOnError(err, "Failed to create shared volume")
+		log.Infof("Shared volume %v created successfully", volumeID)
+		volumeIDs = append(volumeIDs, volumeID)
+
+		for _, directory := range directories {
+			jobID := fmt.Sprintf("fio-%s", directory)
+			jobIDs = append(jobIDs, jobID)
+			fioJob := client.CreateFioJobSpec(volumeID, jobID, directory)
+			err = client.CreateJob(fioJob)
```
Comment (lsrinivas-pure) [inline tests/basic/nomad_test.go:622]: can we move this creating fio job to a common func?

File: tests/basic/nomad_test.go [hunk 3]
```diff
@@ -413,3 +466,798 @@ var _ = Describe("{RunFioJobWithSnapshotOperationsOnNomad}", func() {
...
+
+	It("Should reboot a node, wait for it to come back, and validate FIO jobs", func() {
+		nodes, err := client.ListNodes()
+		log.FailOnError(err, "Failed to list nodes")
+		if len(nodes) > 0 {
+			node := nodes[0]
+			log.Infof("Rebooting node: %v", node.Name)
+			rebootCmd := `reboot`
+			_, err = client.ExecCommandOnNodeSSH(node.ID, rebootCmd)
+			if err == nil {
```
Comment (lsrinivas-pure) [inline tests/basic/nomad_test.go:983]: if check not required here

Comment (dbhatnagar-px) [reply tests/basic/nomad_test.go:983]: No the check is if there is no err then we fail - as the command we are executing is reboot of node - so we have to do this negative check

Comment (dbhatnagar-px) [reply tests/basic/nomad_test.go:983]: We are expecting an error here

File: tests/basic/nomad_test.go (hunk 1 above)
Comment (dbhatnagar-px) [reply tests/basic/nomad_test.go]: Done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**